### PR TITLE
[DAM-008] auto reload after image XHR upload; rename to Upload Asset; etc.

### PR DESCRIPTION
**Pull request by sdbradley**

### CODE REVIEW

**Comment by aneekam** [comment]
https://app.clubhouse.io/enginecommerce/story/1719/as-an-ecommerce-operator-i-can-use-images-uploaded-to-the-digital-asset-manager-when-creating-or-modifying-product-details

**Comment by fastjames** [comment]
I pushed a few spec fixes, one of which involves making the image creation in `DigitalAssetsController` conditional on image params being supplied.

**Comment by sdbradley** [comment]
wonder if it should return a `422` @fastjames in the case where `image_params` is not `present?` ?

File: app/controllers/spree/admin/digital_assets_controller.rb [hunk 1]
```diff
@@ -11,11 +11,18 @@ def index
       end
 
       def create
-        @object.assign_attributes(permitted_resource_params)        
+        @object.assign_attributes(permitted_resource_params)
         if @object.folder.nil?
           @object.folder = spree_folder
         end
         if @object.save
+          if params[:image] && image_params.present?
```
Comment (fastjames) [inline app/controllers/spree/admin/digital_assets_controller.rb:19]: @sdbradley yes, this should probably return a 422 if the image params are not supplied. I was going by the spec, which does not supply those params in the create test.

**Comment by fastjames** [comment]
Part of this feature that was really confusing to me was the exact nature of the relationship between `Spree::DigitalAsset` and `Spree::Asset` (and consequently `Spree::Image`). When uploading a DigitalAsset and creating an Image based on it, we were setting the `type` attribute of the `Image` to `Spree::DigitalAsset`. However, there is a separate table (`spree_digital_assets`) for objects in that class. This resulted in very strange results when attempting to handle the Images tied to these DigitalAssets, since they could not be accessed as Spree::Image objects at all. I backed out those changes just now, and I'm hoping to see more normal behavior.

**Comment by fastjames** [comment]
Closing due to inactivity. We can revisit these when we re-prioritize the DAM feature.

**Pull request closed without merging**